src/utils/openLink.js

```diff
@@ -13,3 +16,21 @@ export function openLinkEmbedded(url: string): void {
 export function openLinkExternal(url: string): void {
   Linking.openURL(url);
 }
+
+export function shouldUseInAppBrowser(browser: BrowserPreference): boolean {
+  if (browser === 'default') {
+    return Platform.OS === 'android';
+  } else {
+    return browser === 'embedded';
+  }
+}
+
+export function openLinkWithUserPreference(url: string, getState: GetState): void {
```

Comment:
(waiting on @gnprice to weigh in here)

Reply:
Hmm. I agree that this interface taking `getState` feels odd. I feel like the direction I'd rather go in is to separate this function from being concerned with our Redux store at all -- that feels to me like a layer that lives above this one.

Probably the simplest resolution is to have this just take a `BrowserPreference`. The caller is already looking around at various other bits of the state, and it's no trouble for it to to pick out that one too and pass it here.

OTOH if we had more call sites, it would start to feel weird that part of the interface of this function was effectively that you had to say something like `getSettings(state).browser` in its second argument. The exact bit of settings that this depends on feels like it should be an internal implementation detail.

Maybe the best solution is a point in between: have it take a `SettingsState`. Then:
* The exact setting it needs is an internal detail, just like in the current version.
* But it's no longer exposed to the whole Redux state.
* Also the fact that it does depend on *some* user setting becomes explicit in the interface. I think it's good for that to be explicit -- I wouldn't e.g. want this to take just one argument `url: string`, and go find the Redux store as a global or something, or if it did then I'd at least want that connection to be drawn clearly in its jsdoc.